x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataType.java

```diff
@@ -223,6 +228,53 @@ public static Collection<DataType> types() {
         return TYPES;
     }
 
+    /**
+     * The types that are valid in function parameters. This is used by the
+     * function tests to enumerate all possible parameters to test error messages
+     * for invalid combinations.
+     */
+    public static Stream<DataType> validFunctionParameters() {
```

Comment:
This method looks oddly specific, and shouldn't be used in productive code from what I understand (As functions already check the params they receive).
So why not moving it to `AbstractScalarFunctionTestCase`? And maybe describe there why are we using this instead of trying with all the types (Related to the other comment)

Reply:
I had it there at first and moved it. I think I'll put it back there, yeah.

I was thinking it could be useful to `assert` that we're only implementing functions for types in this list, but I'm not sure that's worth it. And, if we do it, we can move it.